plugins/provisioners/ansible/config/guest.rb

```diff
@@ -37,7 +37,11 @@ def validate(machine)
         protected
 
         def check_path(machine, path, test_args, error_message_key = nil)
-          remote_path = Pathname.new(path).expand_path(@provisioning_path)
+          if (/cygwin|mswin|mingw|bccwin|wince|emx/ =~ RUBY_PLATFORM) != nil
+            remote_path = File.join(@provisioning_path, path)
+          else
+            remote_path = Pathname.new(path).expand_path(@provisioning_path)
+          end
```

Comment:
This patch won't work with an absolute path (e.g. `/myapp/playbook.yml`), and I am personally not sure that the `RUBY_PLATFORM` condition is a good way to go (`RUBY_PLATFORM` is only [used once](https://github.com/mitchellh/vagrant/blob/b721eb62cfbfa93895d0d4cf019436ab6b1df05d/lib/vagrant/util/subprocess.rb#L133) so far in the whole vagrant code base, and with a different purpose).

I think that we'd better use the same approach as in #542 and do the following:

```ruby
        def check_path(machine, path, test_args, error_message_key = nil)
          remote_path = File.expand_path(path, @provisioning_path)
          
          # Remove drive letter if running on a Windows host
          remote_path = remote_path.gsub(/^[a-zA-Z]:/, "")

          ...
```
Note that `Pathname.new(path).expand_path(@provisioning_path)` is replaced by `File.expand_path(path, @provisioning_path)` for performance and stability reasons (see https://github.com/mitchellh/vagrant/pull/6529#discussion_r45272058).